iclib/ltc6810.py

```diff
@@ -0,0 +1,622 @@
+
+    @classmethod
+    def get_voltage(cls, data_bytes: list[int]) -> float:
+        """Parse the voltage from data bytes.
+
+        The data bytes is expected to be of length ``2``.
+
+        On Page 24, the datasheet states that each LSB is equivalent to
+        100µV. The full range of 16 bytes is from -0.8192V to +5.73V,
+        but negative values are rounded to zero.
+
+        :param data_bytes: The voltage data bytes (of length ``2``).
+        :return: The voltage value.
+        """
+        assert len(data_bytes) == 2
+
+        data = data_bytes[1] << 8 | data_bytes[0]
+
+        return data / 10000
+
+    @classmethod
+    def get_packet_error_code_bytes(
+            cls,
+            data_bytes: Iterable[int],
+    ) -> tuple[int, int]:
+        """Generate packet error code (PEC) from data.
+
+        Refer to Page 55 of LTC6810 datasheet.
+
+        :return: The packet error code bytes.
+        """
+        PEC = [False] * 15
+        PEC[4] = True
+
+        for data_byte in data_bytes:
+            for i in range(7, -1, -1):
+                DIN = bool(data_byte & (1 << i))
+
+                IN0 = DIN ^ PEC[14]
+                IN3 = IN0 ^ PEC[2]
+                IN4 = IN0 ^ PEC[3]
+                IN7 = IN0 ^ PEC[6]
+                IN8 = IN0 ^ PEC[7]
+                IN10 = IN0 ^ PEC[9]
+                IN14 = IN0 ^ PEC[13]
+
+                PEC[14] = IN14
+                PEC[13] = PEC[12]
+                PEC[12] = PEC[11]
+                PEC[11] = PEC[10]
+                PEC[10] = IN10
+                PEC[9] = PEC[8]
+                PEC[8] = IN8
+                PEC[7] = IN7
+                PEC[6] = PEC[5]
+                PEC[5] = PEC[4]
+                PEC[4] = IN4
+                PEC[3] = IN3
+                PEC[2] = PEC[1]
+                PEC[1] = PEC[0]
+                PEC[0] = IN0
+
+        PEC0 = lsb_bits_to_byte(*PEC[7:])
+        PEC1 = lsb_bits_to_byte(*PEC[:7]) << 1
+
+        return PEC0, PEC1
+
+    @classmethod
+    def get_address_poll_command_bytes(
+            cls,
+            address: int,
+            command: int,
+            poll_data_byte_count: int,
+    ) -> list[int]:
+        """Get address poll command bytes.
+
+        Refer to Page 60 in the Datasheet.
+
+        :param address: The device address.
+        :param command: The command.
+        :param poll_data_byte_count: The number of data bytes to be
+                                     polled.
+        :return: The address poll command bytes.
+        """
+        command_bytes = cls.get_address_command_bytes(address, command)
+        poll_data_bytes = ((1 << 8) - 1,) * poll_data_byte_count
+
+        return list(
+            chain(
+                command_bytes,
+                cls.get_packet_error_code_bytes(command_bytes),
+                poll_data_bytes,
+            ),
+        )
+
+    @classmethod
+    def get_address_write_command_bytes(
+            cls,
+            address: int,
```

Comment:
Is including unused param "address" here intentional (i.e. to maintain consistent formatting for command functions?)

Reply:
Good catch, it's a mistake. I am supposed to use it in the code.